overlord/snapstate/sequence/sequence.go

```diff
@@ -180,3 +180,28 @@ func (snapSeq *SnapSequence) RemoveComponentForRevision(snapRev snap.Revision, c
 	// might be nil
 	return unlinkedComp
 }
+
+// ComponentSideInfoForRev returns cref's component side info for the revision
+// (sequence point) indicated by revIdx if there is one.
+func (snapSeq *SnapSequence) ComponentSideInfoForRev(revIdx int, cref naming.ComponentRef) *snap.ComponentSideInfo {
```

Comment:
we might want to produce a nicer panic if revIdx is out of bound?

Reply:
Not sure if I get this one, do you mean the panic in `CurrentComponentSideInfo`? or that this method should panic?

Reply:
just wondering if in the case if we have bug, hopefully caught by tests, were the revIdx is out of bound, wheher producing a panic here with some more context is useful before or if  the runtime one from [revIdx ]is good enough?